### PR TITLE
#27500 Prepare CatalogRule module Tests for PHPUnit 8

### PR DESCRIPTION
**Pull request by lbajsarowicz**
### Description (*)
Preparing PHPUnit Tests to run with PHPUnit 8.
PRs for each module is separated to make tests easier to merge.

### Fixed Issues (if relevant)
1. magento/magento2#27500

### Manual testing scenarios (*)
N/A

### Questions or comments
PR #27701 needs to be merged first (otherwise PHPUnit tests fail)

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

File: app/code/Magento/CatalogRule/Test/Unit/Model/Indexer/ReindexRuleProductPriceTest.php [hunk 1]
```diff
@@ -104,8 +104,8 @@ public function testExecute()
             'product_id' => 100,
             'website_id' => 1,
             'customer_group_id' => 2,
-            'from_time' => mktime(0, 0, 0, date('m'), date('d') - 100),
-            'to_time' => mktime(0, 0, 0, date('m'), date('d') + 100),
+            'from_time' => mktime(0, 0, 0, (int)date('m'), (int)date('d') - 100),
```
Comment (ihor-sviziev) [inline app/code/Magento/CatalogRule/Test/Unit/Model/Indexer/ReindexRuleProductPriceTest.php:107]: Could you explain why it's needed?

Comment (lbajsarowicz) [reply app/code/Magento/CatalogRule/Test/Unit/Model/Indexer/ReindexRuleProductPriceTest.php:107]: I was asked to introduce `strict_types`. 
- `date(...): string` https://www.php.net/manual/en/function.date.php
- `mktime (int, int, int, int, int, int, int) : int` https://www.php.net/manual/en/function.mktime.php

Answer yourself :-)

Comment (ihor-sviziev) [reply app/code/Magento/CatalogRule/Test/Unit/Model/Indexer/ReindexRuleProductPriceTest.php:107]: ah. ok. got it.

**Review by ihor-sviziev — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @ihor-sviziev, thank you for the review.
[ENGCOM-7352](https://jira.corp.magento.com/browse/ENGCOM-7352) has been created to process this Pull Request

**Pull request merged**